assets/src/stories-editor/components/higher-order/with-snap-targets.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * WordPress dependencies
+ */
+import { withSelect } from '@wordpress/data';
+import { compose, createHigherOrderComponent } from '@wordpress/compose';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+
+/**
+ * Internal dependencies
+ */
+import { STORY_PAGE_INNER_HEIGHT, STORY_PAGE_INNER_WIDTH } from '../../constants';
+import { getBlockInnerElement, getRelativeElementPosition } from '../../helpers';
+import { withSnapContext } from '../contexts/snapping';
+
+/**
+ * Higher-order component that returns snap targets for the current block.
+ *
+ * Snap targets are divided into horizontal and vertical ones, and are based on the following data:
+ *
+ * - The page's width / height constraints and center.
+ * - The block's position in relation to its siblings.
+ *
+ * The snap targets are stored in a map with the snap target as the key and the snap lines to be shown as the value.
+ */
+const applyWithSelect = withSelect( ( select, { clientId } ) => {
+	const {
+		getBlocksByClientId,
+		getBlockRootClientId,
+		getBlock,
+		getBlockOrder,
+	} = select( 'core/block-editor' );
+	const { getCurrentPage } = select( 'amp/story' );
+
+	const parentBlock = getBlockRootClientId( clientId );
+
+	const defaultData = {
+		horizontalSnaps: [],
+		verticalSnaps: [],
+		parentBlockElement: null,
+	};
+
+	if ( getCurrentPage() !== parentBlock ) {
+		return defaultData;
+	}
+
+	const parentBlockElement = getBlockInnerElement( getBlock( parentBlock ) );
+
+	if ( ! parentBlockElement ) {
+		return defaultData;
+	}
+
+	const siblings = getBlocksByClientId( getBlockOrder( parentBlock ) )
+		.filter( ( { clientId: blockId } ) => blockId !== clientId )
+		.filter( getBlockInnerElement );
```

Comment:
You extract the inner element in the loop anyway, so this could be extended to:

```suggestion
		.map( getBlockInnerElement )
		.filter( Boolean );
```

And then simplify lines 115-116 and 151-152.

Reply:
True. I think I already did that at some point, not sure why I didn't keep it 🤔

Reply:
Fixed in dc8f7d18c58e887881a0cb405b14a48a7928771c